x/lockup/keeper/lock_test.go

```diff
@@ -144,17 +144,47 @@ func (suite *KeeperTestSuite) TestUnlockPeriodLockByID() {
 	suite.Require().Len(locks, 0)
 }
 
+func (suite *KeeperTestSuite) TestLock() {
+	// test for coin locking
+	suite.SetupTest()
+
+	addr1 := sdk.AccAddress([]byte("addr1---------------"))
+	coins := sdk.Coins{sdk.NewInt64Coin("stake", 10)}
+	lock := types.NewPeriodLock(1, addr1, time.Second, suite.Ctx.BlockTime().Add(time.Second), coins)
+
+	// try lock without balance
+	err := suite.App.LockupKeeper.Lock(suite.Ctx, lock)
+	suite.Require().Error(err)
+
+	// lock with balance
+	suite.FundAcc(addr1, coins)
+	err = suite.App.LockupKeeper.Lock(suite.Ctx, lock)
+	suite.Require().NoError(err)
+
+	// lock with balance with same id
+	suite.FundAcc(addr1, coins)
+	err = suite.App.LockupKeeper.Lock(suite.Ctx, lock)
+	suite.Require().Error(err)
+
+	// lock with balance with different id
+	lock = types.NewPeriodLock(2, addr1, time.Second, suite.Ctx.BlockTime().Add(time.Second), coins)
+	suite.FundAcc(addr1, coins)
+	err = suite.App.LockupKeeper.Lock(suite.Ctx, lock)
+	suite.Require().NoError(err)
+}
+
```

Comment:
should we start writing test that checks gas at every intermediate steps of lock functions? Is that even possible?

Reply:
I really like this idea, want to use this for e2e testings or state compatibility testing :)

Reply:
No this seems excessive. Let's just fix the reason that this was an issue

Reply:
Depends on if we have agreement for this approach: https://github.com/cosmos/cosmos-sdk/issues/12788

Reply:
agreeing on that & finishing tests in https://github.com/osmosis-labs/cosmos-sdk/pull/304 << writing tooling for this lol.